DS13/code/modules/overmap/combat.dm

```diff
@@ -0,0 +1,506 @@
+				var/sound/S = pick('DS13/sound/effects/voice/LoadingPhoton.ogg','DS13/sound/effects/voice/LoadingPhoton2.ogg')
+				voice_alert(S)
+			return
+
+/obj/structure/overmap/proc/special_fx(var/shields_absorbed)
+	if(!main_overmap)
+		special_fx_targeted(shields_absorbed)
+		return
+	var/area = pick(GLOB.teleportlocs)
+	var/area/target = GLOB.teleportlocs[area] //Pick a station area and yeet it.
+	for(var/mob/player in GLOB.player_list)
+		if(is_station_level(player.z))
+			shake_camera(player, 1,3)
+			if(shields_absorbed)
+				if(prob(50))
+					var/sound/shieldhit = pick('DS13/sound/effects/damage/shield_hit.ogg','DS13/sound/effects/damage/shield_hit2.ogg')
+					SEND_SOUND(player, shieldhit)
+				continue
+			var/sound/S = pick('DS13/sound/effects/damage/shiphit.ogg','DS13/sound/effects/damage/shiphit2.ogg','DS13/sound/effects/damage/shiphit3.ogg','DS13/sound/effects/damage/creak1.ogg','DS13/sound/effects/damage/creak2.ogg')
+			SEND_SOUND(player, S)
+	var/max = rand(1,5)
+	for(var/I = 0, I < max, I ++)
+		var/obj/machinery/X = pick(GLOB.machines)
+		X.explode_effect()
+	if(shields_absorbed)
+		return
+	if(prob(40))
+		var/turf/T = pick(get_area_turfs(target))
+		new /obj/effect/temp_visual/explosion_telegraph(T)
+		var/turf/TT = pick(get_area_turfs(target))
+		TT.atmos_spawn_air("plasma=20;TEMP=1000")
+
+/obj/effect/temp_visual/explosion_telegraph
+	name = "Explosion imminent!"
+	icon = 'DS13/icons/effects/effects.dmi'
+	icon_state = "explosion_telegraph"
+	duration = 30
+	randomdir = 0
+	alpha = 0
+	layer = ABOVE_MOB_LAYER
+
+/obj/effect/temp_visual/explosion_telegraph_non_explosive
+	name = "RUN"
+	icon = 'DS13/icons/effects/effects.dmi'
+	icon_state = "explosion_telegraph"
+	duration = 30
+	randomdir = 0
+	layer = ABOVE_MOB_LAYER
+
+/obj/effect/temp_visual/explosion_telegraph/Initialize()
+	. = ..()
+	for(var/turf/T in orange(src, 3))
+		new /obj/effect/temp_visual/explosion_telegraph_non_explosive(T)
+	for(var/mob/F in orange(src, 3))
+		if(isliving(F))
+			to_chat(F, "<span class='userdanger'>Something comes crashing down from the ceiling above you! GET OUT OF THE WAY!</span>")
+			SEND_SOUND(F, 'sound/effects/seedling_chargeup.ogg')
+
+/obj/effect/temp_visual/explosion_telegraph/Destroy()
+	var/turf/T = get_turf(src)
+	explosion(T,3,4,4)
+	. = ..()
+
+/obj/structure/overmap/take_damage(var/atom/source, var/amount = 10)
+	. = ..()
+	var/fluff = rand(1,6)
+	switch(fluff)
+		if(1)
+			visible_message("<span class='warning'>Bits of [name] fly off into space!</span>")
+		if(2)
+			visible_message("<span class='warning'>[name]'s hull ruptures!</span>")
+		if(3)
+			visible_message("<span class='warning'>[name]'s hull buckles!</span>")
+		if(4)
+			visible_message("<span class='warning'>Warp plasma spews from [name]!</span>")
+		if(5)
+			visible_message("<spanR class='warning'>A beam tears across [name]'s hull!</span>")
+		if(6)
+			visible_message("<span class='warning'>[name]'s hull is scorched!</span>")
+	if(!isnum(amount))
```

Comment:
return a debug if this occurs. We should work out why it happens with torps and fix it.

Reply:
Aka the more info the better